plugins/modules/keycloak_user.py

```diff
@@ -0,0 +1,505 @@
+        choices: [ "present", "absent" ]
+        default: present
+        type: str
+    force:
+        description:
+            - If true, allows to remove user and recreate it.
+        type: bool
+        default: false
+extends_documentation_fragment:
+    - community.general.keycloak
+    - community.general.attributes
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: full
+notes:
+    - The module does not modify the user ID of an existing user.
+author:
+    - Philippe Gauthier (@elfelip)
+'''
+
+EXAMPLES = '''
+- name: Create a user user1
+  community.general.keycloak_user:
+    auth_keycloak_url: http://localhost:8080/auth
+    auth_username: admin
+    auth_password: password
+    realm: master
+    username: user1
+    firstName: user1
+    lastName: user1
+    email: user1
+    enabled: true
+    emailVerified: false
+    credentials:
+        - type: password
+          value: password
+          temporary: false
+    attributes:
+        - name: attr1
+          values:
+            - value1
+          state: present
+        - name: attr2
+          values:
+            - value2
+          state: absent
+    groups:
+        - name: group1
+          state: present
+    state: present
+
+- name: Re-create a User
+  community.general.keycloak_user:
+    auth_keycloak_url: http://localhost:8080/auth
+    auth_username: admin
+    auth_password: password
+    realm: master
+    username: user1
+    firstName: user1
+    lastName: user1
+    email: user1
+    enabled: true
+    emailVerified: false
+    credentials:
+        - type: password
+          value: password
+          temporary: false
+    attributes:
+        - name: attr1
+          values:
+            - value1
+          state: present
+        - name: attr2
+          values:
+            - value2
+          state: absent
+    groups:
+        - name: group1
+          state: present
+    state: present
+    force: yes
+
+- name: Remove User.
+  community.general.keycloak_user:
+    auth_keycloak_url: http://localhost:8080/auth
+    auth_username: admin
+    auth_password: password
+    realm: master
+    username: user1
+    state: absent
+'''
+
+RETURN = '''
+user:
+  description: JSON representation for the user.
+  returned: on success
+  type: dict
+msg:
+  description: Message if it is the case
+  returned: always
+  type: str
+changed:
+  description: Return True if the operation changed the client on the keycloak server, false otherwise.
+  returned: always
+  type: bool
+'''
+from ansible_collections.community.general.plugins.module_utils.identity.keycloak.keycloak import KeycloakAPI, camel, \
+    keycloak_argument_spec, get_token, KeycloakError, is_struct_included
+from ansible.module_utils.basic import AnsibleModule
+import copy
+
+
+def main():
+    argument_spec = keycloak_argument_spec()
+    keycloak_override_spec = dict(
+        auth_username=dict(type='str', aliases=[]),
```

Comment:
Don't do this. Simply do `argument_spec['auth_username']['aliases'] = []`. This avoids potentially breaking other things the module_utils relies on.